### PR TITLE
Add option to toggle auto focus on the preview window.

### LINKED ISSUE #44: Configuration: auto-focus and auto-hide of preview windows
Thanks for implementing a great plugin! This might be a question or feature request (if missing).

First, how can I prevent the cursor from automatically being moved to the preview window? I would want to "peek" the definition only, not disturbing the cursor on the main buffer. This should be a configurable option for all the methods, such as `goto_preview_definition(opts)`.

Second, I would want to have the preview window automatically closed and dismissed when the cursor moves (assuming the cursor hasn't be taken to the preview window). This is similar to [vim-quickui](https://github.com/skywind3000/vim-quickui)'s `persist = 0` option. How can I achieve that? Writing a custom `post_open_hook` function for adding autocmd events would be definitely possible, but is there any chance you'd want to implement this is a built-in option?

### PR DESCRIPTION
**Pull request by rmagatti**
Closes #44

### CODE REVIEW

File: lua/goto-preview/lib.lua [hunk 1]
```diff
@@ -101,23 +135,38 @@ local open_floating_win = function(target, position)
...
+    else
+      return M.conf.dismiss_on_move
+    end
+  end
 
+  logger.debug("dismiss_on_move", dismiss())
+  if dismiss() then
+    vim.api.nvim_command(
+      string.format("autocmd CursorMoved <buffer> ++once lua require('goto-preview').dismiss_preview(%d)", new_window)
+    )
```
Comment (rmagatti) [inline lua/goto-preview/lib.lua:153]: Important bit

File: lua/goto-preview/lib.lua [hunk 2]
```diff
@@ -66,33 +68,65 @@ end
...
+    if focus_on_open ~= nil then
+      return focus_on_open
+    else
+      return M.conf.focus_on_open
+    end
+  end
+
+  logger.debug("dismiss_on_move", enter())
+
+  local new_window = vim.api.nvim_open_win(buffer, enter(), {
```
Comment (rmagatti) [inline lua/goto-preview/lib.lua:114]: Important bit

**Comment by wookayin** [comment]
Thanks! I feel the argument (false, true) doesn't seem good to me; we could do this with a table like this:

    ({focus_on_open = false,
      dismiss_on_move = true})

Let's keep the open-closed principle in mind. Can we have this fixed asap to avoid breaking changes in the future as much as possible for users?

**Comment by wookayin** [comment]
@rmagatti Submitted PR #47.

**Pull request merged**